src/v3/src/transformer/selectAuthenticator/utils.ts

```diff
@@ -360,6 +376,30 @@ const formatAuthenticatorOptions = (
         AUTHENTICATOR_KEY.PHONE,
       ];
 
+      // @ts-ignore TODO: Add grace period fields to auth-js SDK https://oktainc.atlassian.net/browse/OKTA-848910
+      const dateMs = new Date(authenticator?.gracePeriod?.expiry).getTime();
+      const gracePeriodEpochTimestampMs = Number.isNaN(dateMs) ? 0 : dateMs;
+      const currentTimestampMs = Date.now();
+
+      let remainingGracePeriodDays = 0;
+      let hasGracePeriods = false;
+      if (currentTimestampMs < gracePeriodEpochTimestampMs) {
+        remainingGracePeriodDays = TimeUtil.calculateDaysBetween(
+          currentTimestampMs,
+          gracePeriodEpochTimestampMs,
+        );
+        hasGracePeriods = true;
+      }
+
+      const gracePeriodExpiry = (hasGracePeriods && locale
+        && TimeUtil.formatDateToDeviceAssuranceGracePeriodExpiryLocaleString(
+          new Date(gracePeriodEpochTimestampMs),
+          locale,
+          false,
+        )) || undefined;
+      const gracePeriodRequiredDescription = (gracePeriodExpiry
+        && getGracePeriodRequiredDescription(remainingGracePeriodDays)) || undefined;
```

Comment:
Generally it's best to use `null` to represent intentionally missing values as opposed to `undefined` to represent unintentionally missing values, unless there's a reason we need to specifically expect `undefined`

Reply:
hmm @jakebacher-okta was there a specific reason to have `undefined` here? I'm guessing no, but just wanted to check before updating to `null`

Reply:
Updated to `null`